google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/SchemaCompactTest.java

```diff
@@ -72,56 +93,907 @@ public Builder toBuilder() {
+                SFixed64Type.getDescriptor(),
+                EnumType.getDescriptor()));
+
+    for (Descriptors.Descriptor descriptor : type_descriptors) {
+      if (compatible.contains(descriptor)) {
+        compact.check("projects/p/datasets/d/tables/t", descriptor, false);
+      } else {
+        try {
+          compact.check("projects/p/datasets/d/tables/t", descriptor, false);
+          fail("Should fail: Proto schema type should not match BQ Timestamp.");
+        } catch (IllegalArgumentException expected) {
+          assertEquals(
+              "The proto field "
+                  + descriptor.getName()
+                  + ".test_field_type does not have a matching type with the big query field t.test_field_type.",
+              expected.getMessage());
+        }
+      }
+    }
+    verify(mockBigquery, times(16)).getTable(any(TableId.class));
+    verify(mockBigqueryTable, times(16)).getDefinition();
+  }
+
+  /*
+   * Tests if having no matching fields in the top level causes an error.
+   */
+  @Test
+  public void testBQTopLevelMismatch() {
+    customizeSchema(
+        Schema.of(
+            Field.newBuilder("test_toplevel_mismatch", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.NULLABLE)
+                .build()));
+    SchemaCompact compact = SchemaCompact.getInstance(mockBigquery);
+    try {
+      compact.check("projects/p/datasets/d/tables/t", StringType.getDescriptor(), false);
+    } catch (IllegalArgumentException expected) {
+      assertEquals(
+          "There is no matching fields found for the proto schema "
+              + StringType.getDescriptor().getName()
+              + " and the BQ table schema t.",
+          expected.getMessage());
+    }
+    verify(mockBigquery, times(1)).getTable(any(TableId.class));
+    verify(mockBigqueryTable, times(1)).getDefinition();
+  }
+
+  /*
+   * Tests if there is at least 1 matching field in the top level.
+   */
+  @Test
+  public void testBQTopLevelMatch() {
+    Field nestedMessage0 =
+        Field.newBuilder("mismatch", LegacySQLTypeName.STRING).setMode(Field.Mode.NULLABLE).build();
+    customizeSchema(
+        Schema.of(
+            Field.newBuilder("mismatch", LegacySQLTypeName.RECORD, nestedMessage0)
+                .setMode(Field.Mode.NULLABLE)
+                .build(),
+            Field.newBuilder("match", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.NULLABLE)
+                .build()));
+    SchemaCompact compact = SchemaCompact.getInstance(mockBigquery);
+    compact.check("projects/p/datasets/d/tables/t", TopLevelMatch.getDescriptor(), false);
+    verify(mockBigquery, times(1)).getTable(any(TableId.class));
+    verify(mockBigqueryTable, times(1)).getDefinition();
+  }
+
+  @Test
+  public void testAllowUnknownUnsupportedFields() {
+    customizeSchema(
+        Schema.of(
+            Field.newBuilder("string_value", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.NULLABLE)
```

Comment:
Add a test case that has a incompatible field which is going to be ignored.

Reply:
The last test case testAllowUnknownUnsupportedFields has a BQ schema of one string value and a proto schema of one string value and one int32 value. The int32 value will be ignored -- is this what you meant by having an incompatible field? My current code logic is that if a field in the BQ schema and proto schema has a matching name, its types must be compatible or a IllegalArgumentException will be thrown.

Reply:
I see. Never mind, just missed the case.